mono/mini/interp/interp.c

```diff
@@ -685,16 +685,21 @@ stackval_to_data_addr (MonoType *type_, stackval *val)
 static MONO_NEVER_INLINE void
 interp_throw (ThreadContext *context, MonoException *ex, InterpFrame *frame, gconstpointer ip, gboolean rethrow)
 {
+	ERROR_DECL (error);
 	MonoLMFExt ext;
 
 	interp_push_lmf (&ext, frame);
 	frame->ip = ip;
 	frame->ex = ex;
 
-	if (!rethrow) {
-		ex->stack_trace = NULL;
-		ex->trace_ips = NULL;
+	if (mono_object_isinst_checked ((MonoObject *) ex, mono_defaults.exception_class, error)) {
+		MonoException *mono_ex = (MonoException *) ex;
```

Comment:
Cheaper to check rethrow before mono_object_isinst_checked?

Reply:
I agree, but I wanted to keep it in sync with https://github.com/mono/mono/blob/5136a60cf4d57820e9d319f0e19dd7f1abcb9da1/mono/mini/exceptions-amd64.c#L389-L395